opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+// Geneva Config Client with TLS (P12) and TODO: Managed Identity support
+
+use base64::{engine::general_purpose, Engine as _};
+use reqwest::Client;
+use serde::Deserialize;
+use std::time::Duration;
+use thiserror::Error;
+use uuid::Uuid;
+
+use chrono::{DateTime, Utc};
+use native_tls::{Identity, Protocol};
+use std::fs;
+use std::sync::RwLock;
+
+/// Authentication methods for the Geneva Config Client.
+///
+/// The client supports two authentication methods:
+/// - Certificate-based authentication using PKCS#12 (.p12) files
+/// - Managed Identity (Azure) - planned for future implementation
+///
+/// # Certificate Format
+/// Certificates should be in PKCS#12 (.p12) format for client TLS authentication.
+///
+/// ## Converting from PEM to PKCS#12
+///
+/// If you have PEM format cert and key, you can convert them using OpenSSL:
+///
+/// ### Linux/macOS:
+/// ```bash
+/// openssl pkcs12 -export \
+///   -in cert.pem \
+///   -inkey key.pem \
+///   -out client.p12 \
+///   -name "alias"
+/// ```
+///
+/// ### Windows (PowerShell):
+/// ```powershell
+/// openssl pkcs12 -export -in cert.pem -inkey key.pem -out client.p12 -name "alias"
+/// ```
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) enum AuthMethod {
+    /// Certificate-based authentication
+    ///
+    /// # Arguments
+    /// * `path` - Path to the PKCS#12 (.p12) certificate file
+    /// * `password` - Password to decrypt the PKCS#12 file
+    Certificate { path: String, password: String },
```

Comment:
Use [`Path`](https://doc.rust-lang.org/std/path/struct.Path.html) instead of `String`?
```suggestion
    Certificate { path: Path, password: String },
```

Reply:
Done